cgrok/grok.go

```diff
@@ -28,6 +28,7 @@ const (
 
 type Grok struct {
 	g *C.grok_t
+	stringCacheLock sync.RWMutex
```

Comment:
Concurrent accesses to this map should be synchronized. After the first match we should have 0 cache misses - RWMutex is pretty cheap for read-only access with no contention.